l10n_nl_tax_statement/models/account_move_line.py

```diff
@@ -31,10 +31,52 @@ def _get_l10n_nl_vat_statement_protected_fields(self):
             "tax_tag_ids",
         ]
 
+    def check_field_is_equal(self, changed_protected_field, values):
+        self.ensure_one()
+        if self._fields.get(changed_protected_field) and self._fields[
```

Comment:
What happens if a non x2m field in the database (existing value) has a falsey value (0.0 or empty string)? From the code I think it will fail the first condition of the if, and therefore proceed to the part where it checks the x2m field. Or am I mistaken in what happens here?

Reply:
I think the code can be more clear if you first have a simple condition for the type of fields and then handle regular fields and x2m fields separately.

Reply:
> What happens if a non x2m field in the database (existing value) has a falsey value (0.0 or empty string)? From the code I think it will fail the first condition of the if, and therefore proceed to the part where it checks the x2m field. Or am I mistaken in what happens here?

It will not continue, because the first return always works. 
We first compare all non-m2m and non-o2m, and return that result.

the second return activates only if the field is m2m  or o2m.


As for the 0.0 , empty string case , automatic typecasts seemed to work well , except for "None":
```
>>> 0==False
True
>>> 0.0==False
True
>>> ''==False
False
>>> ''==0-0
False
>>> ''==0.0
False
>>> None==False
False
```

I will explicitly convert them in "Bool" to make this more solid:
```
   >>> bool('')==bool(0.0)
True
>>> bool(None)==bool(0.0)
True
>>> bool(None)==bool(False)
True
```